linkerd/main/src/main/scala/io/buoyant/linkerd/Main.scala

```diff
@@ -64,10 +64,16 @@ object Main extends App {
   private def initAdmin(
     config: Linker.LinkerConfig,
     linker: Linker
-  ): Closable with Awaitable[Unit] = {
+  ): Seq[Closable with Awaitable[Unit]] = {
     val server = linker.admin.serve(this, LinkerdAdmin(config, linker))
     log.info(s"serving ${linker.admin.scheme} admin on ${server.boundAddress}")
-    server
+
+    config.admin.flatMap(_.httpIdentifierPort) match {
+      case Some(p) =>
+        val idServer = linker.admin.serveHandler(new InetSocketAddress(p.port), LinkerdAdmin.identifierHandler(config, linker))
```

Comment:
I'm glad you're thinking about UX here. I don't see this as internal vs. user complexity. More like accuracy vs. accessibility. I share your concerns though; I don't really see a path towards integrating this with the admin ui.
Serving from the existing admin port would preclude being able to implement this for h2 identifiers. Multiple ports allow for multiple protocols.
Let's talk through it more though, there could be something I'm not thinking of.